src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -101,13 +99,62 @@ public Section getNextSection(Section section) {
     }
 
     private void insertion(Section newSection) {
-        findByUpStation(newSection.getUpStation())
-                .ifPresent(section -> section.shiftBack(newSection));
-        findByDownStation(newSection.getDownStation())
-                .ifPresent(section -> section.shiftForward(newSection));
+        findSectionByUpStationOrDownStation(newSection)
+                .ifPresent(section -> section.shift(newSection));
+
+        if (hasAnyMatchedStation(newSection)) {
+            sections.add(newSection);
+            return;
+        }
+
+        throw new NotMatchedStationException(newSection.getUpStation(), newSection.getDownStation());
+    }
+
+    private boolean hasAnyMatchedStation(Section section) {
+        if (sections.isEmpty()) {
+            return true;
+        }
+        return sections.stream()
+                .anyMatch(mySection -> mySection.anyMatched(section));
+    }
+
+    private Optional<Section> findSectionByUpStationOrDownStation(Section section) {
+        Optional<Section> upStation = findByUpStation(section.getUpStation());
+        if (upStation.isPresent()) {
+            return upStation;
+        }
+        return findByDownStation(section.getDownStation());
     }
 
     public List<Section> getSections() {
         return sections;
     }
+
+    public void deleteSectionByStation(Station station) {
+        validateDeleteSection(station);
+        findByDownStation(station)
+                .ifPresent(section -> {
+                    Section nextSection = getNextSection(section);
+                    mergeIntoNextSection(section, nextSection);
+                    sections.remove(section);
+                });
+        findByUpStation(station)
+                .ifPresent(section -> sections.remove(section));
```

Comment:
하나의 라인에 하나의 점을 써주시면 가독성이 올라갑니다 😄 

```suggestion
        findByDownStation(station).ifPresent(section -> {
            Section nextSection = getNextSection(section);
            mergeIntoNextSection(section, nextSection);
            sections.remove(section);
        });
        findByUpStation(station).ifPresent(section -> sections.remove(section));
```

Reply:
넵 코딩컨벤션 원칙을 지키도록 하겠습니다~!